### PR TITLE
chore: Replace react notebook library with notebookjs

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

As discussed last month, we want to replace our file browser / Jupyter notebook preview dependency on `react-ipynb-renderer` ([link](https://github.com/righ/react-ipynb-renderer)) with a lighter, commonly-used, and up-to-date library.  We previously discussed server-side rendering in Python or Go, but the easier replacement may be finding a small client-side library, which we'll maintain on npm.

I considered:
- forking `react-ipynb-renderer` and replacing the module's package.json for fewer dependencies - Apache license- I tried this some time ago at https://github.com/mapmeld/react-ipynb-renderer but didn't see a significant change in our package-lock.json? Maybe I measured this the wrong way
- nbpreview: https://github.com/jsvine/nbpreview - adds code highlighting, sanitizing, etc to notebookjs, MIT license
- notebookjs: https://github.com/jsvine/notebookjs - based on issue comments, I made a fork and removed `jsdom` dependency to avoid SharedArrayBuffer issue. MIT license. This is super lightweight and we need to find our own CSS, code highlighting, and it's a bit awkward placing it inside of React.  But it does the job of preview

Screen Shot 2022-10-18 at 1 33 40 PM

## Test Plan

Based on latest-master, `localhost:3000/det/experiments/1169/code?filePath=Iris-TensorFlow-Keras.ipynb`

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.
- [x] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

### CODE REVIEW

File: webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx [hunk 1]
```diff
@@ -1,36 +1,19 @@
...
-        katex: {
-          delimiters: 'gitlab', // dollars by default
-          katexOptions: { fleqn: false },
-        },
-        // optional
-        renderer: 'mathjax',
+    <div
+      className="ipynb-renderer-root"
+      dangerouslySetInnerHTML={{
+        __html: file && NotebookJS.parse(JSON.parse(file)).render().outerHTML,
```
Comment (trentwatt) [inline webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx:15]: do you know whether `NotebookJS` sanitizes? apparently `dangerouslySetInnerHTML` can cause scripts to be executed if not escaped properly.

I tried a quick test by dropping a 
```<p onmouseover='alert(`oops`);'>Hi!</p>```
into the notebook and it did the right thing. 

in the process though i saw that a failure of the JSON parse on this line crashes the page (same thing though happens on [master](http://latest-master.determined.ai:8080/det/experiments/1487/code?filePath=Iris-TensorFlow-Keras.ipynb)

**Review by trentwatt — APPROVED** [review]
A win in terms of the bundle size, still think we should work on the styling though -- it doesn't look consistent with the rest of the site.

couple comments below.

**Comment by mapmeld** [comment]
on the style, there are several styles available in https://github.com/righ/react-ipynb-renderer/tree/master/src/styles with the best non-dark-mode version most likely being 'monokai'. I did see that we used the onedork.css in the existing version

**Comment by trentwatt** [comment]
uhh, i guess we could write a script to replace all the colors with the nearest `var(--theme-variables)` in rgb space?? that way we could get a dark and light mode?

File: webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx (hunk 1 above)
Comment (ClaireNeveu) [reply webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx:16]: @mapmeld Did you test this?

Comment (mapmeld) [reply webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx:16]: According to https://pragmaticwebsecurity.com/articles/spasecurity/react-xss-part2.html the best option for `dangerouslySetInnerHTML` is to use a sanitizer such as dompurify.  I looked and that's what's used by notebookjs: (e.g. in `el.innerHTML = nb.sanitizer(nb.markdown(joined))`)

We're limited in what we can do while still inserting arbitrary notebooks into React.

Comment (mapmeld) [reply webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx:16]: I took a closer look at this; if it's helpful I can remove support for JS and SVG cells which don't do the escape.

**Review by ClaireNeveu — APPROVED** [review]

File: webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx (hunk 1 above)
Comment (ClaireNeveu) [reply webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx:16]: Let's remove support for JS, you can leave SVG
